OpenRA.Mods.Common/Traits/Player/BaseAttackNotifier.cs

```diff
@@ -60,12 +60,27 @@ public void Damaged(Actor self, AttackInfo e)
 			if (e.Attacker == self.World.WorldActor)
 				return;
 
+			// This player won't hear us anyway
+			// Unless we want to notify his allies
+			if (!self.Owner.Playable && info.AllyNotification != null)
+				return;
+
+			// Only track last hit against our base
+			if (!self.Info.HasTraitInfo<BuildingInfo>())
+				return;
+
 			if (e.Attacker.Owner.IsAlliedWith(self.Owner) && e.Damage <= 0)
 				return;
 
 			if (self.World.WorldTick - lastAttackTime > info.NotifyInterval * 25)
 			{
-				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", info.Notification, self.Owner.Faction.InternalName);
+				var rules = self.World.Map.Rules;
+				Game.Sound.PlayNotification(rules, self.Owner, "Speech", info.Notification, self.Owner.Faction.InternalName);
+
+				if (info.AllyNotification != null)
```

Comment:
I sometimes use is `string.IsNullOrEmpty`, but I don't think it is required here. Whitespaces are stripped from the fieldloader so that shouldn't cause a problem.